cpp/models/abm/person.h

```diff
@@ -70,127 +49,141 @@ class Person
-     * @param global_infection_parameters infection parameters that are the same in all locations
+     * @brief Time passes and the person interacts with the population at its current location.
+     * The person might become infected.
+     * @param[in] t Current time.
+     * @param[in] dt Length of the current simulation time step.
+     * @param[in,out] global_infection_parameters Infection parameters that are the same in all locations.
      */
-    void interact(TimeSpan dt, const GlobalInfectionParameters& global_infection_parameters, Location& loc);
+    void interact(TimePoint t, TimeSpan dt, GlobalInfectionParameters& params);
 
     /** 
-     * migrate to a different location.
-     * @param loc_new the new location of the person.
-     * @param cells_new the new cells of the person.
+     * @brief Migrate to a different location.
+     * @param[in] loc_new The new location of the person.
+     * @param[in] cells_new The new cells of the person.
      * */
-    void migrate_to(Location& loc_old, Location& loc_new, const std::vector<uint32_t>& cells_new = {});
+    void migrate_to(Location& loc_new, const std::vector<uint32_t>& cells_new = {0});
 
     /**
-     * Get the current infection state of the person.
-     * @returns the current infection state of the person
+     * @brief Get the latest Infection of the Person.
+     * @returns The latest Infection of the Person.
      */
-    InfectionState get_infection_state() const
+    Infection& get_infection()
     {
-        return m_infection_state;
+        return m_infections.back();
     }
 
-    /**
-     * Get the current vaccination state of the person.
-     * @returns the current vaccination state of the person
-     */
-    VaccinationState get_vaccination_state() const
+    const Infection& get_infection() const
+    {
+        return m_infections.back();
+    }
+
+    /** 
+     * @returns All vaccinations.
+    */
+    std::vector<Vaccination>& get_vaccinations()
+    {
+        return m_vaccinations;
+    }
+
+    const std::vector<Vaccination>& get_vaccinations() const
     {
-        return m_vaccination_state;
+        return m_vaccinations;
     }
 
     /**
-     * Sets the current infection state of the person.
-     */
-    void set_infection_state(InfectionState inf_state);
+     * @brief Returns if the person is infected at the time point.
+     * @param[in] t Time point of querry. Usually the current time of the simulation.
+     * @returns True if the person is infected at the time point.
+    */
+    bool is_infected(TimePoint t) const;
+
+    /**
+     * @param[in] t Time point of querry. Usually the current time of the simulation.
+     * @returns The infection state of the latest infection at time t.
+    */
+    InfectionState get_infection_state(TimePoint t) const;
+
+    /**
+     * @brief Adds a new infection to the list of infections.
+     * @param[in] inf The new infection.
+    */
+    void add_new_infection(Infection&& inf);
```

Comment:
It makes sense to have the rvalue version then, since in most cases, the infection is created and then immediately added to the person and not needed anymore. If we need other overloads, they can be added later.